drivers/ad9208/ad9208.h

```diff
@@ -0,0 +1,157 @@
+#include "platform_drivers.h"
+
+#define AD9208_FULL_BANDWIDTH_MODE 0
+#define AD9208_1_DDC_MODE 1
+#define AD9208_2_DDC_MODE 2
+#define AD9208_4_DDC_MODE 4
+
+#define AD9208_SYSREF_NONE 0	/* No SYSREF Support */
+#define AD9208_SYSREF_ONESHOT 1	/* ONE-SHOT SYSREF */
+#define AD9208_SYSREF_CONT 2	/* Continuous Sysref Synchronisation */
+#define AD9208_SYSREF_MON 3	/* SYSREF monitor Mode */
+
+#define AD9208_NCO_MODE_VIF 0	/* Variable IF Mode */
+#define AD9208_NCO_MODE_ZIF 1	/* Zero IF Mode */
+#define AD9208_NCO_MODE_TEST 3	/* Test Mode*/
+
+#define AD9208_BUFF_CURR_400_UA  0x4	/* Buffer Current set to 400 uA */
+#define AD9208_BUFF_CURR_500_UA  0x9	/* Buffer Current set to 500 uA */
+#define AD9208_BUFF_CURR_600_UA  0x1E	/* Buffer Current set to 600 uA */
+#define AD9208_BUFF_CURR_700_UA  0x23	/* Buffer Current set to 700 uA */
+#define AD9208_BUFF_CURR_800_UA  0x28	/* Buffer Current set to 800 uA */
+#define AD9208_BUFF_CURR_1000_UA  0x32	/* Buffer Current set to 1000 uA */
+
+#define AD9208_CHIP_TYPE	0x03
+#define AD9208_CHIP_ID		0xDF
+
+struct ad9208_ddc {
+	u32 decimation;
+	u32 nco_mode;
+	u64 carrier_freq_hz;
+	u64 po;
+	bool gain_db;
+};
+
+typedef struct ad9208_dev {
+	/* SPI */
+	spi_desc *spi_desc;
+	/* GPIO */
+	gpio_desc *gpio_powerdown;
+	struct ad9208_state *st;
+} ad9208_dev;
+
+struct ad9208_state {
+	ad9208_handle_t *adc_h;
+	uint64_t sampling_frequency_hz;
+	uint32_t input_div; /* input clock divider ratio */
+	bool powerdown_pin_en;
+	uint32_t powerdown_mode;
+	bool duty_cycle_stabilizer_en;
+	uint8_t current_scale;
+	bool analog_input_mode;
+	bool ext_vref_en;
+	uint32_t buff_curr_n;
+	uint32_t buff_curr_p;
+	uint8_t fc_ch;
+	struct ad9208_ddc ddc[4];
+	uint32_t ddc_cnt;
+	bool ddc_output_format_real_en;
+	bool ddc_input_format_real_en;
+	uint32_t test_mode_ch0;
+	uint32_t test_mode_ch1;
+
+	uint32_t sysref_lmfc_offset;
+	bool sysref_edge_sel;
+	bool sysref_clk_edge_sel;
+	uint32_t sysref_neg_window_skew;
+	uint32_t sysref_pos_window_skew;
+	uint32_t sysref_mode;
+	uint32_t sysref_count;
+
+	jesd_param_t *jesd_param;
+	uint32_t jesd_subclass;
+};
+
+typedef struct ad9208_init_param {
+	/* SPI */
+	spi_init_param *spi_init;
+	/* GPIO */
+	int8_t gpio_powerdown;
+	uint64_t sampling_frequency_hz;
+	uint32_t input_div; /* input clock divider ratio */
+	bool powerdown_pin_en;
+	uint32_t powerdown_mode;
+	bool duty_cycle_stabilizer_en;
+	uint8_t current_scale;
+	bool analog_input_mode;
+	bool ext_vref_en;
+	uint32_t buff_curr_n;
+	uint32_t buff_curr_p;
+	uint8_t fc_ch;
+	struct ad9208_ddc *ddc;
+	uint32_t ddc_cnt;
+	bool ddc_output_format_real_en;
+	bool ddc_input_format_real_en;
+	uint32_t test_mode_ch0;
+	uint32_t test_mode_ch1;
+
+	uint32_t sysref_lmfc_offset;
+	bool sysref_edge_sel;
+	bool sysref_clk_edge_sel;
+	uint32_t sysref_neg_window_skew;
+	uint32_t sysref_pos_window_skew;
+	uint32_t sysref_mode;
+	uint32_t sysref_count;
+
+	jesd_param_t *jesd_param;
+	uint32_t jesd_subclass;
+} ad9208_init_param;
+
```

Comment:
/******************************************************************************/
/************************ Functions Declarations ******************************/
/******************************************************************************/

/*! Initialize the device. */